index.js

```diff
@@ -159,7 +159,10 @@ Shopify.prototype.request = function request(uri, method, key, data, headers) {
             limit: this.options.maxRetries,
             // Don't clamp Shopify `Retry-After` header values too low.
             maxRetryAfter: Infinity,
-            calculateDelay
+            calculateDelay,
+            methods: [method],
+            statusCodes: Array.from(retryableStatusCodes),
```

Comment:
Instead of having separate retryable codes/errors between `got` and `shopify-api-node`, we should pass our expected retryable codes and errors to `got`

Reply:
I'm not a fan of this. Copying the array for every request does not seem a good idea. The same applies to the line below.

Reply:
I will update this to be a `const` at the module level a re-use the same array for each call